File: src/app/models/SensorInstance.js


// Define collection and schema for SensorInstance
export interface SensorInstance {
    name:
	type : string
    unit:
	type : string
    samplingIntervalMs:
	type : number
    Device:
	type : Schema.Types.ObjectId
    TelemetryStreams:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TelemetryStream' }]
    SensorType:
 	type : String
#
    collection: 'sensorInstances'
}

File: iotonaspdotnet/Domain/SensorInstance.cs

using iotonaspdotnet.Contracts;

namespace iotonaspdotnet.Domain;

public class SensorInstance
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? SensorinstanceId { get; set; }
    public virtual string? Name { get; set; }
    public virtual string? Unit { get; set; }
    public virtual int? SamplingIntervalMs { get; set; }
    public virtual IoTDevice? Device { get; set; }
    public virtual ICollection<TelemetryStream> TelemetryStreams { get; set; } = new List<TelemetryStream>();
    public virtual SensorType? SensorType { get; set; }

    public static SensorInstance FromRequest(SensorInstanceRequest request)
    {
        return new SensorInstance
        {
            Id = request.Id,
            Name = request.Name,
            Unit = request.Unit,
            SamplingIntervalMs = request.SamplingIntervalMs,
            SensorType = request.SensorType,
        };
    }
}
